scripts/gen-nncp.sh

```diff
@@ -315,17 +315,61 @@ EOF_CAT
     fi
 
     #
-    # octavia-vlan-link VLAN interface
+    # storage_mgmt VLAN interface
     #
     cat >> ${DEPLOY_DIR}/${WORKER}_nncp.yaml <<EOF_CAT
-    - description: octavia-vlan-link vlan interface
+    - description: storage_mgmt vlan interface
       name: ${INTERFACE}.$((${VLAN_START}+$((${VLAN_STEP}*3))))
```

Comment:
This is swapping the VLAN ID's around as well?


In `scripts/gen-netatt.sh`:
```
      "name": "octavia",
      "type": "macvlan",
      "master": "${INTERFACE}.$((${VLAN_START}+${VLAN_STEP}*3))",

      "name": "designate",
      "type": "macvlan",
      "master": "${INTERFACE}.$((${VLAN_START}+${VLAN_STEP}*4))",
```


Now we are inserting storage_mgmt at `${VLAN_STEP}*3` and shifting octavia and disignate? But that is not updated in all the other files?

I think we should add the `storage_mgmt` network at the end and use ${VLAN_STEP}*5?

Reply:
yes, I put it there to match the vlan id from the standalone deployment, where storage_mgmt has vlan 23. We can either change the vlans in `scripts/gen-netatt.sh` or the standalone network config to keep the current order and add `storage_mgmt` at the end as you suggest. Judging by the ip ranges, the storage_mgmt seemed like it should go after the `tenant` network and before the octavia one

Reply:
IDK - it seems like we have a complete mess here?

The `NetConfig` defines `storagemgmt` as VLAN 23.
https://github.com/openstack-k8s-operators/infra-operator/blob/main/config/samples/network_v1beta1_netconfig.yaml#L45-L53

I think that is the one used based on `Makefile:NETCONFIG     ?= config/samples/network_v1beta1_netconfig.yaml`?

But in NNCP we are currently using 23 for `octavia`?

so probably changing octavia to use a diferent VLAN is the best way to go? I think that would be only changes in install_yamls repo `gen-nncp.sh` and `gen-netatt.sh`?

Reply:
yes, many moving parts here, let me update the patch to set the same vlan ids in gen-nncp and net-netatt